test/libsolidity/SolidityExecutionFramework.cpp

```diff
@@ -31,22 +31,46 @@ using namespace solidity::frontend;
 using namespace solidity::frontend::test;
 using namespace std;
 
+bytes SolidityExecutionFramework::multiSourceCompileContract(
+	map<string, string> const& _sourceCode,
+	string const& _contractName,
+	map<string, Address> const& _libraryAddresses
+)
+{
+	map<string, string> sourceCodeMap = _sourceCode;
+	for (auto &entry: sourceCodeMap)
+		addPreamble(entry.second);
+
+	return compileContractCommon(
+		sourceCodeMap,
+		_contractName,
+		_libraryAddresses
+	);
+}
+
 bytes SolidityExecutionFramework::compileContract(
 	string const& _sourceCode,
 	string const& _contractName,
 	map<string, Address> const& _libraryAddresses
 )
 {
-	// Silence compiler version warning
-	std::string sourceCode = "pragma solidity >=0.0;\n";
-	if (
-		solidity::test::CommonOptions::get().useABIEncoderV2 &&
-		_sourceCode.find("pragma experimental ABIEncoderV2;") == std::string::npos
-	)
-		sourceCode += "pragma experimental ABIEncoderV2;\n";
-	sourceCode += _sourceCode;
+	string sourceCode = _sourceCode;
+	addPreamble(sourceCode);
```

Comment:
If you don't know that `addPreamble()` modifies its argument this makes you stop and think why a new variable is being introduced. I think a more explicit name would make this much clearer:
```suggestion
	string sourceCodeWithPreamble = _sourceCode;
	addPreamble(sourceCodeWithPreamble);
```

Reply:
This is no longer relevant after other changes.